portal-api/src/cron-scheduler-jobs/helpers/utilisation-report-helpers.ts

```diff
@@ -0,0 +1,203 @@
+  return month === getOneIndexedMonth(previousMonthDate);
+};
+
+/**
+ * Sends the email to the specified bank when a valid payment officer team email is present
+ * @param emailDescription {string} - a description of the email (used for logging)
+ * @param sendEmailCallback {SendEmailCallback} - callback function that sends the required email.
+ *   The callback is provided the following argument `{ emailAddress: string; recipient: string }`,
+ *   extracted from the provided bank.
+ * @param bank {object} - the bank to send the email to
+ * @returns {Promise<void>}
+ */
+const sendEmailForBank = async (
+  emailDescription: string,
+  formattedReportPeriod: string,
+  sendEmailCallback: SendEmailCallback,
+  bank: BankResponse,
+): Promise<void> => {
+  const { name: bankName, paymentOfficerTeam, id: bankId } = bank;
+  try {
+    const paymentOfficerTeamEmails = paymentOfficerTeam?.emails;
+
+    if (!Array.isArray(paymentOfficerTeamEmails) || !paymentOfficerTeamEmails.length) {
+      console.warn(`Not sending ${emailDescription} email to '${bankName}' - paymentOfficerTeam.emails property against bank is not an array or is empty`);
+      return;
+    }
+    await Promise.all(
+      paymentOfficerTeamEmails.map(async (paymentOfficerTeamEmail) => {
+        if (!isValidEmail(paymentOfficerTeamEmail)) {
+          console.error(`Failed to send ${emailDescription} email to '${paymentOfficerTeamEmail}' - invalid payment officer email`);
+        } else {
+          await sendEmailCallback(paymentOfficerTeamEmail, getEmailRecipient(paymentOfficerTeam, bankName), formattedReportPeriod);
+          console.info(`Successfully sent '${emailDescription}' email to '${bankName}' (bank ID: ${bankId})`);
+        }
+      }),
+    );
+  } catch (error) {
+    console.error(`Failed to send ${emailDescription} email for bank '${bankName}':`, error);
+  }
+};
+
+/**
+ * For a given bank, check if a utilisation report has been received for the current reporting period
+ * then, where not yet received, attempts to call the provided callback function to send the required email
+ * @param emailDescription {string} - a description of the email (used for logging)
+ * @param sendEmailCallback {SendEmailCallback} - callback function that sends the required email.
+ *   The callback is provided the following argument `{ emailAddress: string; recipient: string }`,
+ *   extracted from the provided bank.
+ * @returns {Promise<void>}
+ */
+const sendEmailToBankIfReportNotReceived = async (bank: BankResponse, emailDescription: string, sendEmailCallback: SendEmailCallback): Promise<void> => {
+  const { name: bankName, id: bankId, utilisationReportPeriodSchedule: schedule } = bank;
+  const currentReportingPeriodForBank = getCurrentReportPeriodForBankSchedule(schedule);
+  const formattedReportPeriod = getFormattedReportPeriod(currentReportingPeriodForBank);
+
+  if (isPreviousCalendarMonth(currentReportingPeriodForBank.end.month)) {
```

Comment:
Could we swap around the `if-else` here to return early if `isPreviousCalendarMonth` returns `false`?

Reply:
Good idea! Originally I couldn't because this was part of the function below but now it is extracted that is a good shout